File: CloudflareFileStorage.js
const fs = require('fs');
const axios = require('axios');

class CloudflareFileStorage {
    constructor(workerUrl) {
        this.workerUrl = workerUrl.replace(/\/$/, '');
    }

    async uploadFile(filePath) {
        if (!fs.existsSync(filePath)) {
            throw new Error(`File not found: ${filePath}`);
        }

        const fileContent = fs.createReadStream(filePath);
        const formData = new FormData();
        formData.append('file', fileContent, { filename: path.basename(filePath) });

        const response = await axios.post(`${this.workerUrl}/upload`, formData, {
            headers: {
                ...formData.getHeaders(),
                'Content-Type': 'application/octet-stream',
            },
        });

        return response.data;
    }

    async downloadFile(fileName, savePath) {
        const response = await axios.get(`${this.workerUrl}/download/${fileName}`, { responseType: 'arraybuffer' });
        if (response.status !== 200) {
            throw new Error(`Error downloading file: ${response.statusText}`);
        }

        const filePath = path.join(savePath, fileName);
        fs.writeFileSync(filePath, response.data);
        return response.data;
    }

    async deleteFile(fileName) {
        const response = await axios.delete(`${this.workerUrl}/delete/${fileName}`);
        return response.data;
    }
}

File: CloudflareFileStorage.py
import os
import requests

class CloudflareFileStorage:
    def __init__(self, worker_url):
        self.worker_url = worker_url.rstrip('/')
    
    def upload_file(self, file_path):
        if not os.path.exists(file_path):
            raise FileNotFoundError(f"File not found: {file_path}")

        with open(file_path, 'rb') as file:
            files = {'file': (os.path.basename(file_path), file, 'application/octet-stream')}
            response = requests.post(f"{self.worker_url}/upload", files=files)
            return response.text
    
    def download_file(self, file_name, save_path):
        response = requests.get(f"{self.worker_url}/download/{file_name}")
        if response.status_code != 200:
            raise Exception(f"Error downloading file: {response.reason}")

        file_path = os.path.join(save_path, file_name)
        with open(file_path, 'wb') as file:
            file.write(response.content)
        return response.content
    
    def delete_file(self, file_name):
        response = requests.delete(f"{self.worker_url}/delete/{file_name}")
        return response.text
